api/v1alpha1/openapi.yaml

```diff
@@ -3833,15 +3833,38 @@ components:
           type: string
           description: An opaque string that identifies the server's internal version of an object.
       description: ObjectMeta is metadata that all persisted resources must have, which includes all objects users must create.
+
+    MatchExpression:
+      properties:
+        key:
+          type: string
+        operator:
+          type: string
+          enum: [In, NotIn, Exists, DoesNotExist]
+        values:
+          type: array
+          items:
+            type: string
+      required:
+        - key
+        - operator
+
+    MatchExpressions:
+      type: array
+      minItems: 1
+      items:
+        $ref: '#/components/schemas/MatchExpression'
+
     LabelSelector:
       type: object
       properties:
         matchLabels:
           type: object
           additionalProperties:
             type: string
-      required:
```

Comment:
It's a bit misleading functionality-wise; it implies that someone can define a LabelSelector but keep both matchLabels and matchExpressions empty.

Reply:
I don't know if there's a better way.  Though we should make sure to validate it in the code (api/v1alpha1/validation.go).

Reply:
Can we keep the matchLabels AS IS and only set matchExpressions as optional?

Reply:
If we keep it as is then we will be required to always add the matchLabels field even if it is empty.